src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs

```diff
@@ -196,6 +204,37 @@ private void LoadExtensions()
             }
         }
 
+        private void InitializeDataCollectors(string runSettings, ITestEventsPublisher testEventsPublisher, IEnumerable<TestCase> tests)
```

Comment:
Code duplicacy. We can just pass defaultCodeBase. Write 2 methods for GetDefaultCodeBase (for testcases and adaptersourcemap)